### PR TITLE
Ensure generated types are properly documented

### LINKED ISSUE #2027: Generated code in release 1.16.0 breaks CS1591
After upgrading to the latest release 1.16.0 we are not able to release build with treat warnings as errors. It seems that the generated code [breaks roslyn rule 1591](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1591)

```
2>GeneratedFunctionExecutor.g.cs(97,25): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "FunctionExecutorHostBuilderExtensions" fehlt.
2>GeneratedFunctionExecutor.g.cs(110,18): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "FunctionExecutorAutoStartup" fehlt.
2>GeneratedFunctionExecutor.g.cs(112,21): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "FunctionExecutorAutoStartup.Configure(IHostBuilder)" fehlt.
2>GeneratedFunctionMetadataProvider.g.cs(14,18): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "GeneratedFunctionMetadataProvider" fehlt.
2>GeneratedFunctionMetadataProvider.g.cs(16,56): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "GeneratedFunctionMetadataProvider.GetFunctionMetadataAsync(string)" fehlt.
2>GeneratedFunctionMetadataProvider.g.cs(160,25): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "WorkerHostBuilderFunctionMetadataProviderExtension" fehlt.
2>GeneratedFunctionMetadataProvider.g.cs(175,18): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "FunctionMetadataProviderAutoStartup" fehlt.
2>GeneratedFunctionMetadataProvider.g.cs(177,21): Error CS1591 : Der XML-Kommentar für den öffentlich sichtbaren Typ oder Member "FunctionMetadataProviderAutoStartup.Configure(IHostBuilder)" fehlt.
```

(this states: xml comment is missing on public visible type )

our Directory.Build.targets look like:

```
<Project>
    <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
        <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
        <TreatSpecificWarningsAsErrors/>
        <WarningsAsErrors>true</WarningsAsErrors>
        <WarningsNotAsErrors>618;$(WarningsNotAsErrors)</WarningsNotAsErrors>
    </PropertyGroup>
</Project>
```

and Directory.Build.props:

```
    <PropertyGroup>
        <AnalysisMode>Recommended</AnalysisMode>
        <TargetFramework>net7.0</TargetFramework>
        <GenerateDocumentationFile>true</GenerateDocumentationFile>
        <PreserveCompilationContext>true</PreserveCompilationContext>
        <LangVersion>latest</LangVersion>
    </PropertyGroup>
```

Having release 1.15.1 release build is not a problem and works flawless.

### PR DESCRIPTION
**Pull request by fabiocav**
Issue describing the changes in this PR

resolves #2027 

The changes ensure public types and members are documented. It also adds `EditorBrowsableAttribute` where appropriate to avoid having confusion with those APIs 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs [hunk 1]
```diff
@@ -42,6 +43,10 @@ public async ValueTask ExecuteAsync(FunctionContext context)
                                      {{GetMethodBody(functions)}}
                                  }
                              }
+
```
Comment (kshyju) [inline sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs:46]: Do we need "/// <inheritdoc/>" in line 40?

Comment (fabiocav) [reply sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs:46]: Not required as this type itself is internal. But will add for consistency.

**Review by kshyju — APPROVED** [review]
Do we need [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/68b972f22059bbfc5e846abae72a6ac6afb48827/sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs#L101) ?

**Comment by fabiocav** [comment]
> Do we need [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/68b972f22059bbfc5e846abae72a6ac6afb48827/sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs#L101) ?

Not needed, but doing the same and updating for consistency.

**Pull request merged**